taskmanager/taskq.go

```diff
@@ -89,76 +96,88 @@ func (c *Client) loadTaskQ() error {
 	return nil
 }
 
-// registerTaskUsingTaskQ will register a new task using the TaskQ engine
-func (c *Client) registerTaskUsingTaskQ(task *Task) {
-
+// RegisterTask will register a new task using the TaskQ engine
+func (c *TaskManager) RegisterTask(name string, handler interface{}) (err error) {
 	defer func() {
-		if err := recover(); err != nil {
-			c.DebugLog(fmt.Sprintf("registering task panic: %v", err))
+		if panicErr := recover(); panicErr != nil {
+			err = fmt.Errorf(fmt.Sprintf("registering task panic: %v", panicErr))
 		}
 	}()
 
 	mutex.Lock()
+	defer mutex.Unlock()
 
-	// Check if task is already registered
-	if t := taskq.Tasks.Get(task.Name); t != nil {
-
-		// Register the task locally
-		c.options.taskq.tasks[task.Name] = t
+	if t := taskq.Tasks.Get(name); t != nil {
+		// if already registered - register the task locally
+		c.options.taskq.tasks[name] = t
+	} else {
+		// Register and store the task
+		c.options.taskq.tasks[name] = taskq.RegisterTask(&taskq.TaskOptions{
+			Name:       name,
+			Handler:    handler,
+			RetryLimit: 1,
+		})
+	}
 
-		// Task already exists!
-		// c.DebugLog(fmt.Sprintf("registering task: %s... task already exists!", task.Name))
+	c.DebugLog(fmt.Sprintf("registering task: %s...", c.options.taskq.tasks[name].Name()))
+	return nil
+}
 
-		mutex.Unlock()
+// RunTask will run a task using TaskQ
+func (c *TaskManager) RunTask(ctx context.Context, options *TaskRunOptions) error {
+	c.options.logger.Info().Msgf("executing task: %s", options.TaskName)
 
-		return
+	// Try to get the task
+	task, ok := c.options.taskq.tasks[options.TaskName]
+	if !ok {
+		return fmt.Errorf("task %s not registered", options.TaskName)
 	}
 
-	// Register and store the task
-	c.options.taskq.tasks[task.Name] = taskq.RegisterTask(convertTaskToTaskQ(task))
+	// Task message will be used to add to the queue
+	taskMessage := task.WithArgs(ctx, options.Arguments...)
 
-	mutex.Unlock()
+	// There are two ways to run a task:
+	// Option 1: Run the task immediately
+	if options.RunEveryPeriod == 0 {
```

Comment:
```suggestion
	if runImmediately(options) {
```

Reply:
I added a method to the `TaskRunOptions` struct. Now the line looks like this. 
```golang
if options.runImmediately() {
		return c.options.taskq.queue.Add(taskMessage)
	}
```